src/plugins/data/public/query/saved_query/saved_query_service.ts

```diff
@@ -105,8 +105,13 @@ export const createSavedQueryService = (
   };
 
   const getSavedQuery = async (id: string): Promise<SavedQuery> => {
-    const savedObject = await savedObjectsClient.get<SerializedSavedQueryAttributes>('query', id);
-    if (savedObject.error) {
+    const {
+      saved_object: savedObject,
+      outcome,
+    } = await savedObjectsClient.resolve<SerializedSavedQueryAttributes>('query', id);
+    if (outcome === 'conflict') {
+      throw new Error(`Multiple saved queries found with ID: ${id}`);
+    } else if (savedObject.error) {
       throw new Error(savedObject.error.message);
     }
     return parseSavedQueryObject(savedObject);
```

Comment:
Just to let you know, when a saved query is not found, this error checking logic won't work as expected.
Currently if you `get` an object with the public SOC, it batches the request into a `bulkGet` under the hood - if it can't find the object, it doesn't throw an error, rather it just returns an object with an `error` field.

The `resolve` API doesn't work the same way, so it will actually throw an error when an object is not found.

_That said_, I have an open issue to add a `bulkResolve` with the same type of transparent batching, so after that is merged the behavior will be back to how you expect it: #111201

TL;DR no changes needed here :+1: